src/GHC/RTS/Events/Binary.hs

```diff
@@ -936,6 +938,9 @@ skipExtra expected get_body = do
 
 -- | Skip extra stuff at the end of a variable sized event
 -- For forwards compatability (allowing newer events to be read with older versions)
+--
+-- The body parser expectes an expected length for the input (stored in the eventlog)
+-- a position of the parser
```

Comment:
Maybe this Haddock comment should be attached to the type signature of the body_parser.

Reply:
If I understand you correctly you mean something like:

```
-- | Skip extra stuff at the end of a variable sized event
-- For forwards compatability (allowing newer events to be read with older versions)
variableSizeParser :: Int
                   -> (Word16 -> Int -> Get a) -- The body parser expectes an expected length for the input (stored in the eventlog)
                                               -- a position of the parser
                   -> EventParser a
```

But then the comment won't show up in the actual haddocks anymore I think?

So I would rather keep it part of the proper haddocks. I did rephrase the comment slightly as after not looking at it for half a year it wasn't as clear as it seemed back then.

Reply:
Seems matt went ahead and made this change in https://github.com/haskell/ghc-events/pull/83.